openmdao/core/system.py

```diff
@@ -1246,6 +1249,57 @@ def list_states(self, stream=sys.stdout):
         else:
             stream.write("\nNo states in %s.\n" % pathname)
 
+    def list_unit_conv(self, stream=sys.stdout):
+        """ List all unit conversions that are being handled by OpenMDAO
+        (including those with units defined only on one side of the
+        connection.)
+
+        Args
+        ----
+        stream : output stream, optional
+            Stream to write the state info to. Default is sys.stdout.
+
+        Returns
+        -------
+            List of unit conversions.
+        """
+
+        params_dict = self._params_dict
+        unknowns_dict = self._unknowns_dict
+        connections = self.connections
+
+        # Find all unit conversions
+        unit_diffs = {}
+        for target, (source, idxs) in iteritems(connections):
+            tmeta = params_dict[target]
+            smeta = unknowns_dict[source]
+
+            # If we have a conversion, there should be a conversion factor
+            # tucked away in the params meta
+            conv = tmeta.get('unit_conv')
```

Comment:
This check will only find cases where both sides of a connection have unit strings.  The cases where only one side has units are missing.